### PR TITLE
add fdopendir

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: src/unix/mod.rs [hunk 1]
```diff
@@ -328,6 +328,10 @@ extern {
     #[cfg_attr(all(target_os = "macos", target_arch = "x86"),
                link_name = "fdopen$UNIX2003")]
     pub fn fdopen(fd: ::c_int, mode: *const c_char) -> *mut ::FILE;
+
+    #[cfg(not(any(target_os = "macos", target_os = "ios")))]
```
Comment (Mic92) [inline src/unix/mod.rs]: Should I define this function here or in every os module except macos?

Comment (alexcrichton) [inline src/unix/mod.rs]: Ah yeah let's just inline this definition in modules other than `src/unix/bsd/apple`

Comment (Mic92) [inline src/unix/mod.rs]: done

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

**Comment by Mic92** [comment]
passes the test now

**Pull request merged**